webui/react/src/utils/trial.ts

```diff
@@ -43,6 +45,45 @@ export const metricNameToValue = (metricName: MetricName): string => {
   return `${metricName.type}|${metricName.name}`;
 };
 
+export const metricNameFromValue = (metricValue: string): MetricName | undefined => {
+  const trainingPrefix = `${MetricType.Training}|`;
+  const validationPrefix = `${MetricType.Validation}|`;
+  if (metricValue.startsWith(trainingPrefix)) {
+    return {
+      name: metricValue.slice(trainingPrefix.length),
+      type: MetricType.Training,
+    };
+  } else if (metricValue.startsWith(validationPrefix)) {
+    return {
+      name: metricValue.slice(validationPrefix.length),
+      type: MetricType.Validation,
+    };
+  } else {
+    const errName = 'metricNameFromValueUnrecognizedMetricType';
+    const errSlug = 'metricnamefromvalue-unrecognized-metric-type';
+    const errMessage = `metricNameFromValue was called, but the metricName doesn't appear to " +
+        'be a training metric or a validation metric (${metricValue})`;
```

Comment:
@hkang1  Is including the `metricValue` as part of the error message too much detail?

Reply:
I think `metricValue` would be great to have since this is more for us to use when figuring out how we got something that is neither training nor validation metric.

Reply:
**suggestion:** I think in template literals, everything will be preserved in the string, including the double quote, plus character and the single quote. You can do something like this and the newlines are typically not a problem unless we are presenting them in an element with a CSS definition of `white-space: pre;`, which is really rare and something we avoid in general.
```
const errMessage = `
  metricNameFromValue was called, but the metricName doesn't appear to
  be a training metric or a validation metric (${metricValue})
`;
```

Reply:
Good catch! I changed it from a normal string to a template string and didn't notice that the quotes/plus sign became part of the string. Fixed